### PR TITLE
feat(analytics): Dart implementation

### PR DESCRIPTION
**Pull request by fjnoyp**
*Issue #, if available:*

*Description of changes:*

Merge analytics in next release branch 

All the changes here have already been reviewed and approved.  

This is the main analytics dart first PR + db_common refactor PR. 

Reference: 

Analytics dart
https://github.com/aws-amplify/amplify-flutter/pull/2197

db_common
https://github.com/aws-amplify/amplify-flutter/pull/2316

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ragingsquirrel3** [comment]
request: Could CI errors be fixed here where possible? Specifically flutter unit test for amplify_flutter https://circleci.com/gh/aws-amplify/amplify-flutter/71220 and format https://circleci.com/gh/aws-amplify/amplify-flutter/71212?

File: packages/analytics/amplify_analytics_pinpoint/example/web/index.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+<html>
+
+<head>
+  <meta charset="UTF-8">
+  <title>amplify_analytics_pinpoint_example_app</title>
+</head>
+
+<body>
+<!-- Added for Drift web setup: https://drift.simonbinder.eu/web/ -->
+<script defer src="sql-wasm.js"></script>
```
[the remark this answers is not in the document]
Comment (ragingsquirrel3) [reply packages/analytics/amplify_analytics_pinpoint/example/web/index.html:11]: Likewise, does it need the wasm file itself sibling here? sql-wasm.wasm

**Comment by fjnoyp** [comment]
Thanks @ragingsquirrel3 I ran a format on the analytics dart file.  Unsure on how unit tests are failing as Analytics doesn't actually introduce any!  I can check more into that.

**Comment by ragingsquirrel3** [comment]
Why was a license file moved to /tool dir? at top-level of repo.

File: packages/amplify/amplify_flutter/lib/src/method_channel_amplify.dart [hunk 2]
```diff
@@ -126,13 +119,4 @@ class MethodChannelAmplify extends AmplifyClassImpl {
       }
     }
   }
-
```
Comment (ragingsquirrel3) [inline packages/amplify/amplify_flutter/lib/src/method_channel_amplify.dart:129]: question: why was this part removed here?

**Comment by fjnoyp** [comment]
I think the LICENSE was added by some build step on accident, I removed it.  

For that line I think if you look at the entire file you'll see that I deleted the reset method that space was there to separate the methods but just below is another closing bracket.

File: packages/amplify/amplify_flutter/lib/src/method_channel_amplify.dart (hunk 2 above)
Comment (ragingsquirrel3) [reply packages/amplify/amplify_flutter/lib/src/method_channel_amplify.dart:129]: talked offline, seems removed bc superclass does same thing

File: packages/analytics/amplify_analytics_pinpoint/pubspec.yaml [hunk 3]
```diff
@@ -9,26 +9,28 @@ environment:
...
+  linux:
+  web:
+
 dependencies:
-  amplify_analytics_pinpoint_android: ">=1.0.0-next.0 <1.0.0-next.1"
-  amplify_analytics_pinpoint_ios: ">=1.0.0-next.0 <1.0.0-next.1"
-  amplify_core: ">=1.0.0-next.0+2 <1.0.0-next.1"
-  aws_common: ">=0.2.4 <0.3.0"
+  amplify_analytics_pinpoint_dart: ^0.1.0
+  amplify_core: ">=1.0.0-next.0+1 <1.0.0-next.1"
```
Comment (ragingsquirrel3) [inline packages/analytics/amplify_analytics_pinpoint/pubspec.yaml:23]: question: Are we sure these deps are correct here from last rebase? seems like other packages get core from .0+2, also noticing the ways of specifying below deps like amplify_secure_storage different than auth.

File: packages/analytics/amplify_analytics_pinpoint_dart/pubspec.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+name: amplify_analytics_pinpoint_dart
+description: A Dart-only implementation of the Amplify Analytics plugin for Pinpoint.
+version: 0.1.0
+homepage: https://docs.amplify.aws/lib/q/platform/flutter/
+repository:
```
Comment (ragingsquirrel3) [inline packages/analytics/amplify_analytics_pinpoint_dart/pubspec.yaml:5]: nit: repo field blank

File: packages/analytics/amplify_analytics_pinpoint_dart/pubspec.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+issue_tracker: https://github.com/aws-amplify/amplify-flutter/issues
+
+environment:
+  sdk: ">=2.17.0 <3.0.0"
+
+dependencies:
+  amplify_core: ^1.0.0-next.0
```
Comment (ragingsquirrel3) [inline packages/analytics/amplify_analytics_pinpoint_dart/pubspec.yaml:12]: question: Should these deps be referred in similar manner as other plugins?

File: packages/analytics/amplify_analytics_pinpoint/pubspec.yaml (hunk 3 above)
Comment (dnys1) [reply packages/analytics/amplify_analytics_pinpoint/pubspec.yaml:23]: They will be updated in version bump - no worries if they're stale here.

File: .github/workflows/lint.yml [hunk 6]
```diff
@@ -172,6 +172,58 @@ jobs:
         if: "always() && steps.packages_common_amplify_db_common_example_pub_upgrade.conclusion == 'success'"
         working-directory: packages/common/amplify_db_common/example
         run: flutter analyze --fatal-infos lib
+      - id: packages_storage_amplify_storage_s3_example_linux_flutter_ephemeral_.plugin_symlinks_amplify_db_common_pub_upgrade
```
Comment (dnys1) [inline .github/workflows/lint.yml:175]: Whoops

File: packages/amplify/amplify_flutter/test/amplify_test.dart [hunk 7]
```diff
@@ -13,7 +13,7 @@
  * permissions and limitations under the License.
  */
 
-import 'package:amplify_analytics_pinpoint/method_channel_amplify.dart';
+import 'package:amplify_datastore/method_channel_datastore.dart';
```
Comment (ragingsquirrel3) [inline packages/amplify/amplify_flutter/test/amplify_test.dart:16]: This test is still failing, will ping offline w suggestions.

**Review by ragingsquirrel3 — APPROVED** [review]
Still some lint errors looks like in CI? Would be good to fix those, seems like in amplify_analytics_pinpoint_dart and just a few things. Otherwise looks good from a sanity check and checking integration of latest fixes as I didn't completely re-review the entire analytics implementation that was already reviewed/approved.

File: packages/analytics/amplify_analytics_pinpoint_dart/pubspec.yaml (hunk 5 above)
Comment (fjnoyp) [reply packages/analytics/amplify_analytics_pinpoint_dart/pubspec.yaml:12]: I think we can disregard like Dillon said and commit to changing this in the version bump PR

File: packages/amplify/amplify_flutter/test/amplify_test.dart (hunk 7 above)
Comment (fjnoyp) [reply packages/amplify/amplify_flutter/test/amplify_test.dart]: Thanks for the fixes

File: .github/workflows/lint.yml (hunk 6 above)
Comment (fjnoyp) [reply .github/workflows/lint.yml]: Thanks reran the mono_repo generate

**Review by dnys1 — APPROVED** [review]
